librz/core/canalysis.c

```diff
@@ -5940,6 +5940,119 @@ RZ_API bool rz_core_analysis_everything(RzCore *core, bool experimental, char *d
 	if (!is_unknown_file(core)) {
 		rz_analysis_add_device_peripheral_map(core->bin->cur->o, core->analysis);
 	}
+
+	if (rz_config_get_b(core->config, "analysis.apply.signature")) {
+		int n_applied = 0;
+		char message[100];
+		rz_print_rowlog(core->print, "Applying signatures from sigdb");
+		rz_core_analysis_sigdb_apply(core, &n_applied, NULL);
+		rz_strf(message, "Applied %d FLIRT signatures via sigdb", n_applied);
+		rz_print_rowlog_done(core->print, message);
+	}
+
+	return true;
+}
+
+/**
+ * \brief Outputs the list of signatures found in the flirt.sigdb.path
+ *
+ * \param core The RzCore instance
+ */
+RZ_API void rz_core_analysis_sigdb_list(RzCore *core) {
+	const char *sigdb_path = rz_config_get(core->config, "flirt.sigdb.path");
```

Comment:
Probably better to separate loading signatures and listing them. Like we do with type databases, for example.

Reply:
If you mean having a `RzList<RzAnalysisSignature> *rz_analysis_sigdb_list` and then a `_print` method that prints that list of structures, then I agree.